local.tf

```diff
@@ -144,10 +144,10 @@ locals {
 
   kubeconfig = var.create_eks ? templatefile("${path.module}/templates/kubeconfig.tpl", {
     kubeconfig_name                   = local.kubeconfig_name
-    endpoint                          = aws_eks_cluster.this[0].endpoint
-    cluster_auth_base64               = aws_eks_cluster.this[0].certificate_authority[0].data
+    endpoint                          = length(aws_eks_cluster.this) > 0 ? aws_eks_cluster.this[0].endpoint : ""
```

Comment:
Please use the splat syntax instead.

`coalescelist(aws_eks_cluster.this[*]. endpoint, [""])[0]` This will return `""` if the state is empty.